applications/GeoMechanicsApplication/custom_operations/apply_excavation_operation.h

```diff
@@ -0,0 +1,113 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Lorenzo Gracia
+//                   Aron Noordam
+//                   Vahid Galavi
+//                   Marjan Fathian
+//                   Ruben Zorrilla
+//
+
+#pragma once
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "operations/operation.h"
+
+// Application includes
+
+namespace Kratos
+{
+
+///@addtogroup GeoMechanicsApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @author Lorenzo Gracia
+ * @author Aron Noordam
+ * @author Vahid Galavi
+ * @author Marjan Fathian
+ * @author Ruben Zorrilla
+ * @class ApplyExcavationOperation
+ * @ingroup GeoMechanicsApplication
+ * @brief This operation emulate the effect of an excavation by deactivating some parts of the computational domain
+ */
+class KRATOS_API(GEO_MECHANICS_APPLICATION) ApplyExcavationOperation : public Operation
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ApplyExcavationOperation
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyExcavationOperation);
+
+    /// Registry current operation
+    KRATOS_REGISTRY_ADD_PROTOTYPE("Operations.KratosMultiphysics.GeoMechanicsApplication", Operation, ApplyExcavationOperation)
+    KRATOS_REGISTRY_ADD_PROTOTYPE("Operations.All", Operation, ApplyExcavationOperation)
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
```

Comment:
We follow this as a standard for the header documentation, so my suggestion would be to keep it. Note that this might be trivial for you, but not for a newbie or a student which is not as acquainted with c++.

Reply:
Okay clear, we can leave it like it is for this PR!